src/_pytest/main.py

```diff
@@ -630,21 +630,15 @@ def _visit_filter(f):
     def _tryconvertpyarg(self, x):
         """Convert a dotted module name to path."""
         try:
-            loader = pkgutil.find_loader(x)
-        except ImportError:
+            spec = importlib.util.find_spec(x)
+        except (ValueError, ImportError):
             return x
-        if loader is None:
+        if spec is None or spec.origin in {None, "namespace"}:
```

Comment:
`getattr(spec, "origin", None)` seems like a good fit here ^^

Reply:
this is ~slightly more explicit as to what's happening here (plus we can factor out the second when going py37+) -- you're right that `getattr(...)` would work here though